config/flags.go

```diff
@@ -0,0 +1,109 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/spf13/pflag"
+	c "nav/constants"
+)
+
+const (
+	defaultMode       = c.PrintSubsys
+	defaultOutputType = "graphOnly"
+	defaultSymbol     = "NEEDED"
+	defaultMaxDepth   = 0
+)
+
+const (
+	appName  string = "Nav - kernel symbol navigator"
+	appUsage string = "Usage:\n  nav [flags]"
+)
+
+func initFlagSet(fs *pflag.FlagSet, configPath *string) {
+	fs.StringVarP(configPath, "config", "f", "", "path to `config` file")
+
+	fs.StringP("symbol", "s", defaultSymbol, "name of the `symbol` to start the navigation from")
+	fs.StringP("output-type", "j", defaultOutputType, "`type` of output: graphOnly, jsonOutputPlain, jsonOutputB64 or jsonOutputGZB64")
+	fs.IntP("max-depth", "x", defaultMaxDepth, "max `number` of levels in call flow exploration (0=No limit)")
+	fs.IntP("mode", "m", int(defaultMode), "`mode` of plotting. 1=Symbols, 2=Subsystems, 3=Subsystems with labels, 4=Target subsystem isolation")
+	fs.StringSliceP("excluded-before", "b", nil, "list of `symbols` to exclude before the target symbol")
+	fs.StringSliceP("excluded-after", "a", nil, "list of `symbols` to exclude after the target symbol")
+	fs.StringSliceP("target-subsys", "t", nil, "list of `subsystems` to include in the output")
+
+	fs.StringP("db-name", "n", "", "target database `name`")
+	fs.StringP("db-url", "d", "", "database `URL`")
+	fs.StringP("db-user", "u", "", "database `user`")
+	fs.StringP("db-password", "p", "", "database `password`")
+	fs.IntP("db-port", "P", 0, "database `port`")
+	fs.IntP("db-instance", "i", 0, "database `instance`")
+
+	fs.BoolP("help", "h", false, "show this help message and exit")
+}
+
+func parseCommandLine(fs *pflag.FlagSet, configPath *string, args []string) error {
+	fs.SortFlags = false // disable sorting so flags get printed in the order they are defined
+	initFlagSet(fs, configPath)
+
+	if err := fs.Parse(args); err != nil {
+		return fmt.Errorf("unable to parse args: %w", err)
+	}
+
+	help, err := fs.GetBool("help")
+	if err != nil {
```

Comment:
Since this is a BoolP I'm not sure what error could possibly occur. The user doesn't say "-h true|false" here, after all. Is this intended to catch bugs, when we accidentally remove the flag form the FlagSet or something like that or just based on the general principle of not ignoring error values?

Reply:
My thinking was more around the general "do not ignore errors" principle :)